daemon/cmd/daemon_main.go

```diff
@@ -1543,6 +1543,13 @@ func initEnv(vp *viper.Viper) {
 			)
 		}
 	}
+
+	// Ensure the xdp-ignore-device-regex compiles
+	if len(option.Config.XDPIgnoreDeviceNameRegex) > 0 {
```

Comment:
Was this question from @squeed ever discussed? Scanning through I could see a few folks saying there should be a discussion but I couldn't see a clear resolution to the question:

> Question for you, @borkmann -- given the usual XDP use-case, would an allowing regex make more sense (i.e. use XDP for devices matching the regex)? Or should we stick with an excluding regex?
> 
> Separately, are there any other specifiers that might make sense? Only interfaces with a given driver?

Reply:
Sorry for the huge delay on my side. First release, and then out for long PTO, it slipped my radar.

Just thinking some more on this topic, @poblahblahblah / @joestringer / @joamaki, what if instead of regex we just had a simple bool option where we say "if XDP attach fails on this device, then just fallback to tc BPF". Would this be more generic and easier to use rather than specifying device regex? Something like `--bpf-lb-acceleration-fallback=true    [ In case BPF load balancing acceleration via native XDP is not supported for some devices, fall back to tc BPF attachment ]`. Other option could be to have a new mode `--bpf-lb-acceleration=best-effort` ?

Reply:
Makes sense to me, no need for the config flag even IMO. If the device is expected to be managed, we should make it so.

Reply:
You mean just to fallback in case of failure with the `native` option? I think this perhaps could lead to false conclusions, meaning people will try to active, then some ethtool driver channel setting is not correct and it'll fail and people just assume XDP would be there now but actually we run in tc BPF mode. So I would bail out by default and let users make a conscious  decision to fallback upon failure.

Reply:
So you are proposing the following:

* Keep the default behavior of bailing out in place
* Implement `--bpf-lb-acceleration=best-effort` which if set would have devices fall back to TC

If I updated this PR to do the above, would the PR be considered mergeable?

Reply:
> If I updated this PR to do the above, would the PR be considered mergeable?

Yes, definitely - if code looks good then I consider it mergable.

Reply:
@borkmann Sorry for the delay. I made a first pass at this change here: https://github.com/cilium/cilium/pull/28666